### PR TITLE
Ci/add python 39 310 311 to GitHub workflows

### PR DESCRIPTION
**Pull request by bpshaver**
## Proposed changes

This PR aims to address [Issue 45](https://github.com/AI-Planning/pddl/issues/45), adding support for Python 3.9, 3.10, and 3.11.

## Fixes

I added Python versions 3.9, 3.10, and 3.11 to the matrix of jobs run by the "test" and "linting" Github workflows. I also added Python 3.8, which looks like it used to be included but was removed at some point.

I also updated `setup.py` to make it clear that Python versions 3.7 -> 3.11 are supported. The `python_requires` argument in `setup.py` will cause installation of `pddl` to fail for Python versions below 3.7. I can remove this line -- I think everything should work with Python 3.6, for example, but that the only reason to remove support for Python 3.6 was that it is technically past its end of life.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

This seemed like a small chore to add these Python versions and ensure that the tests and linting still passed. I hope this meets the requirement!

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - ci/add-python-39-310-311-to-github-workflows
```
Comment (haz) [inline .github/workflows/test.yml:7]: Any reason this branch would need to remain in the testing path?

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - ci/add-python-39-310-311-to-github-workflows
```
Comment (haz) [inline .github/workflows/linting.yml:7]: Any reason this branch would need to remain in the testing path?

**Review by haz — CHANGES_REQUESTED** [review]
Thanks, @bpshaver ! Just one small comment on two spots -- tests seem to run clean, and the changes look straight-forward and correct otherwise.

File: .github/workflows/linting.yml (hunk 2 above)
Comment (bpshaver) [reply .github/workflows/linting.yml]: Oops, that was a mistake. I added my own working branch to test the workflow on my fork...

Just removed this line.

**Review by haz — APPROVED** [review]

**Comment by haz** [comment]
Thank you, @bpshaver !!

**Pull request merged**